impeller/entity/contents/test/recording_render_pass.cc

```diff
@@ -10,65 +10,85 @@ namespace impeller {
 
 RecordingRenderPass::RecordingRenderPass(
     std::shared_ptr<RenderPass> delegate,
-    const std::shared_ptr<Context>& context,
+    const std::shared_ptr<const Context>& context,
     const RenderTarget& render_target)
     : RenderPass(context, render_target), delegate_(std::move(delegate)) {}
 
 // |RenderPass|
 void RecordingRenderPass::SetPipeline(
     const std::shared_ptr<Pipeline<PipelineDescriptor>>& pipeline) {
   pending_.pipeline = pipeline;
-  delegate_->SetPipeline(pipeline);
+  if (delegate_) {
```

Comment:
Why is this all conditional now?

Reply:
This is a test helper, by making it conditional I can make delegating to a real commmand buffer optional.